src/sentry/api/serializers/models/project.py

```diff
@@ -98,13 +97,15 @@ def get_access_by_project(
 ) -> MutableMapping[Project, MutableMapping[str, Any]]:
     request = env.request
 
-    project_teams = list(ProjectTeam.objects.filter(project__in=projects).select_related("team"))
-    project_team_map = defaultdict(list)
+    project_teams = ProjectTeam.objects.filter(project__in=projects).values_list(
+        "project_id", "team_id"
+    )
 
-    for pt in project_teams:
-        project_team_map[pt.project_id].append(pt.team)
+    project_to_teams = defaultdict(list)
+    for project_id, team_id in project_teams:
+        project_to_teams[project_id].append(team_id)
 
-    team_memberships = _get_team_memberships([pt.team for pt in project_teams], user)
+    team_memberships = _get_team_memberships([team_id for _, team_id in project_teams], user)
```

Comment:
you can construct this array in 3 lines before where we already iterate it in a for loop. therefore, we wouldn't iterate the items twice